cargo-shuttle/src/lib.rs

```diff
@@ -337,42 +337,53 @@ impl Shuttle {
                 It will be hosted at ${{project_name}}.shuttleapp.rs, so choose something unique!
                 "
             };
-            let client = self.client.as_ref().unwrap();
-            loop {
-                // not using validate_with due to being blocking
-                let p: String = Input::with_theme(&theme)
+        }
+        let mut prev_name: Option<String> = None;
+        loop {
+            // prompt if interactive
+            let name: String = if let Some(name) = project_args.name.clone() {
+                name
+            } else {
+                // not using `validate_with` due to being blocking.
+                Input::with_theme(&theme)
                     .with_prompt("Project name")
-                    .interact()?;
-                match client.check_project_name(&p).await {
-                    Ok(true) => {
-                        println!("{} {}", "Project name already taken:".red(), p);
-                        println!("{}", "Try a different name.".yellow());
-                    }
-                    Ok(false) => {
-                        project_args.name = Some(p);
-                        break;
-                    }
-                    Err(e) => {
-                        // If API error contains message regarding format of error name, print that error and prompt again
-                        if let Ok(api_error) = e.downcast::<ApiError>() {
-                            // If the returned error string changes, this could break
-                            if api_error.message.contains("Invalid project name") {
-                                println!("{}", api_error.message.yellow());
-                                println!("{}", "Try a different name.".yellow());
-                                continue;
-                            }
-                        }
-                        // Else, the API error was about something else.
-                        // Ignore and keep going to not prevent the flow of the init command.
-                        project_args.name = Some(p);
-                        println!(
-                            "{}",
-                            "Failed to check if project name is available.".yellow()
-                        );
-                        break;
-                    }
+                    .interact()?
+            };
+            // handle forced name on retires
+            let name: String = if let Some(prev_name) = prev_name {
+                if project_args.name.is_none() && name == prev_name {
+                    project_args.name = Some(prev_name);
+                    break;
+                } else {
+                    name
                 }
+            } else {
+                name
+            };
```

Comment:
I'm thinking there should be a way to combine this with the next if since they share logic. Something like

Something like
```rust
let should_force = prev_name.map(|n| n == name).unwrap_or(args.force_name);
```

This is most likely not accurate, but you get the gist :smile:

Reply:
Can't break a loop inside a closure. This anti-pattern is only because `if let  ... = ... && expr` is not stable yet.